plugins/modules/site_workflow_manager.py

```diff
@@ -1276,6 +1387,22 @@ def get_diff_merged(self, config):
                             name_list = [site.get("name") for site in self.want if "name" in site]
                             self.created_site_list.append(name_list)
                             self.log("Site '{}' created successfully".format(name_list), "INFO")
+
+                            for site in self.want:
+                                if site.get("type") == "floor":
+                                    floor_name = site.get("name")
+                                    self.log("Floor '{}' created successfully".format(floor_name), "INFO")
+
+                                    upload_path = site.get("upload_floor_image_path", None)
+                                    if upload_path:
+                                        self.log("Floor '{}' found. Proceeding to upload floor map from '{}.'".format(floor_name, upload_path), "INFO")
+                                        map_details, map_status, success_message = self.upload_floor_image(config)
+                                        if map_details:
+                                            self.log(success_message, "INFO")
+                                        else:
+                                            self.log("Floor map upload failed.", "ERROR")
+                                    else:
+                                        self.log("Floor '{}' created without uploading a floor map.".format(floor_name), "INFO")
```

Comment:
self.log("No upload path provided for '{}'. Floor created without floor map.".format(floor_name), "INFO")

Reply:
Modified log